src/main/java/org/hbase/async/HBaseClient.java

```diff
@@ -1450,10 +1450,8 @@ private AsyncBufferedMutator getBufferedMutator(TableName table) {
     AsyncBufferedMutator mutator = mutators.get(table);
 
     if (mutator == null) {
-      synchronized (mutators) {
-        mutator = hbase_asyncConnection.getBufferedMutator(table);
-        mutators.put(table, mutator);
-      }
+      mutators.putIfAbsent(table, hbase_asyncConnection.getBufferedMutator(table));
+      mutator = mutators.get(table);
```

Comment:
This should likely be something like this:

```java
AsyncBufferedMutator mutator = mutators.get(table);
 if (mutator == null) {
   AsyncBufferedMutator storedMutator = mutators.putIfAbsent(table, hbase_asyncConnection.getBufferedMutator(table));
   if (storedMutator != null){
      mutator.close();
      mutator = storedMutator;
  }
}
return mutator;
```

Reply:
My way has one less look up, and I'd prefer that way, if possible.  `putIfAbsent()` returns the value that's already associated with the key (see the [javadoc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#putIfAbsent(K,%20V))).